spi_loopback test fails on MCUs with SRAM blocks smaller than 128KB

**Describe the bug**
https://github.com/zephyrproject-rtos/zephyr/pull/69081 increases RAM usage by an extra ~100KB, causing .bss section overflows on devices with small memories. I have encountered this on an out-of-tree Arm Cortex M4 board however I expect similar results with other boards.

The PR adds below buffers that I suspect are the cause of overflows.
```c
#define BUF3_SIZE 8192
...
static const char large_tx_data[BUF3_SIZE] = "Thequickbrownfoxjumpsoverthelazydog\0";
static __aligned(32) char large_buffer_tx[BUF3_SIZE] __used __NOCACHE;
static __aligned(32) char large_buffer_rx[BUF3_SIZE] __used __NOCACHE;
...
static uint8_t large_buffer_print_tx[BUF3_SIZE * 5 + 1];
static uint8_t large_buffer_print_rx[BUF3_SIZE * 5 + 1];
``` 

**Expected behavior**
Test should compile successfully.

**Logs and console output**
```
...
.../arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd.exe: region `RAM' overflowed by 58432 bytes
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build
...
```

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain: Zephyr SDK 0.16.3

--- a/tests/drivers/spi/spi_loopback/src/spi.c
+++ b/tests/drivers/spi/spi_loopback/src/spi.c
@@ -88,9 +88,6 @@ static uint8_t buffer_print_rx[BUF_SIZE * 5 + 1];
 static uint8_t buffer_print_tx2[BUF2_SIZE * 5 + 1];
 static uint8_t buffer_print_rx2[BUF2_SIZE * 5 + 1];
 
-static uint8_t large_buffer_print_tx[BUF3_SIZE * 5 + 1];
-static uint8_t large_buffer_print_rx[BUF3_SIZE * 5 + 1];
-
 static void to_display_format(const uint8_t *src, size_t size, char *dst)
 {
 	size_t i;
@@ -558,10 +555,6 @@ static int spi_complete_large_transfers(struct spi_dt_spec *spec)
 	}
 
 	if (memcmp(large_buffer_tx, large_buffer_rx, BUF3_SIZE)) {
-		to_display_format(large_buffer_tx, BUF3_SIZE, large_buffer_print_tx);
-		to_display_format(large_buffer_rx, BUF3_SIZE, large_buffer_print_rx);
-		LOG_ERR("Large Buffer contents are different: %s", large_buffer_print_tx);
-		LOG_ERR("                             vs: %s", large_buffer_print_rx);
 		zassert_false(1, "Large Buffer contents are different");
 		return -1;
 	}
@@ -691,10 +684,6 @@ static int spi_async_call(struct spi_dt_spec *spec)
 	}
 
 	if (memcmp(large_buffer_tx, large_buffer_rx, BUF3_SIZE)) {
-		to_display_format(large_buffer_tx, BUF3_SIZE, large_buffer_print_tx);
-		to_display_format(large_buffer_rx, BUF3_SIZE, large_buffer_print_rx);
-		LOG_ERR("Buffer 3 contents are different: %s", large_buffer_print_tx);
-		LOG_ERR("                             vs: %s", large_buffer_print_rx);
 		zassert_false(1, "Buffer 3 contents are different");
 		return -1;
 	}

